pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -433,11 +436,43 @@ func (c *Reconciler) createBuildPod(ctx context.Context, tr *v1alpha1.TaskRun, t
 	return c.KubeClientSet.CoreV1().Pods(tr.Namespace).Create(pod)
 }
 
+// combineEnvVars takes a base slice of environment variables and a slice of environment
+// variables to overlay on top of the base slice, returning the result.
+func combineEnvVars(base []corev1.EnvVar, override []corev1.EnvVar) []corev1.EnvVar {
```

Comment:
this looks like a good candidate for making a public function in its own package, with its own unit tests :D